dysnix/app/templates/deployment.yaml

```diff
@@ -1,129 +1,219 @@
-{{- if eq .Values.appKind "Deployment" }}
+{{/* vim: set filetype=helm: */}}
+
+{{- define "app.resources.deployment" -}}
+{{- if and .Values.app.workload.enabled (eq .Values.app.workload.type "deployment") }}
+---
 apiVersion: {{ include "common.capabilities.deployment.apiVersion" . }}
 kind: Deployment
 metadata:
-  name: {{ include "common.names.fullname" . }}
-  labels: {{- include "common.labels.standard" . | nindent 4 }}
+  name: {{ template "common.names.fullname" . }}
+  namespace: {{ include "common.names.namespace" . | quote }}
+  labels:
+    {{- include "common.labels.standard" . | nindent 4 }}
+    {{- include "app.labels.component" . | nindent 4 }}
     {{- if .Values.commonLabels }}
     {{- include "common.tplvalues.render" ( dict "value" .Values.commonLabels "context" $ ) | nindent 4 }}
     {{- end }}
-  annotations:
-    {{- if .Values.commonAnnotations }}
-    {{- include "common.tplvalues.render" ( dict "value" .Values.commonAnnotations "context" $ ) | nindent 4 }}
-    {{- end }}
+  {{- if .Values.commonAnnotations }}
+  annotations: {{- include "common.tplvalues.render" ( dict "value" .Values.commonAnnotations "context" $ ) | nindent 4 }}
+  {{- end }}
 spec:
+  {{- if not .Values.autoscaling.enabled }}
   replicas: {{ .Values.replicaCount }}
+  {{- end }}
   {{- if .Values.updateStrategy }}
-  strategy: {{- tpl (toYaml .Values.updateStrategy) $ | nindent 4 }}
+  strategy: {{- toYaml .Values.updateStrategy | nindent 4 }}
   {{- end }}
   selector:
-    matchLabels: {{- include "common.labels.matchLabels" . | nindent 6 }}
+    matchLabels:
+      {{- include "common.labels.matchLabels" . | nindent 6 }}
+      {{- include "app.labels.component" . | nindent 6 }}
   template:
     metadata:
-      {{- if or .Values.checksums .Values.podAnnotations }}
       annotations:
-        {{- range .Values.checksums }}
-        checksum/{{ . | trimPrefix "/" }}: {{ include (print $.Template.BasePath "/" (. | trimPrefix "/")) $ | sha256sum }}
-        {{- end }}
+        {{- include "app.template.checksums" $ | indent 8 }}
         {{- if .Values.podAnnotations }}
-        {{- include "common.tplvalues.render" (dict "value" .Values.podAnnotations "context" $) | nindent 8 }}
-        {{- end }}
-      {{- end }}
-      labels: {{- include "common.labels.standard" . | nindent 8 }}
-        {{- if .Values.commonLabels }}
```

Comment:
Why do we need to remove `commonLabels` from a pod template ?

Reply:
I think commonLabels is not meant for pods, there's podLabels instead.

Reply:
Hm, `commonLabels` should be common :) for all the generated resources, as for me. I threat `podLabels` as extra labels besides `commonLabels`. WDYT on renaming it to `podExtraLabels` in this case? It looks natural to me to "inherit" custom deployment labels as pod labels too. I have to use some duplication with my tests of the new chart version:
```
commonLabels:
  app: {{ .Release.Labels.app }}
podLabels:
  app: {{ .Release.Labels.app }}
```
as I want to select deployments and pods by the same custom label. 

Do you have any examples where it may be useful to have different values of the same label between deployment and it's pods, such as:
```
kind: Deployment
metadata:
  name: foo-release0
  labels:
    app: foo
---
kind: Pod
metadata:
  name: foo-release0-5854854558-dnd9m
  labels:
    app: bar
```
 Otherwise I don't see how `commonLabels` inheritance to pods may hurt.

  We're not talking about `.spec.selector.matchLabels` & related here, just to be clear.